processor/transformprocessor/README.md

```diff
@@ -14,44 +14,131 @@
+The Transform Processor allows configuring OTTL statements for traces, metrics, and logs.
 
-Each context will be processed in the order specified.
-Within a context, each global condition is checked and if any evaluates to true, the statements are executed in order.
-If a context doesn't meet any of the conditions, then the associated statement will be skipped.
+> [!NOTE]
+> If you don't know how to write OTTL statements yet, first see [Writing OTTL](#writing-ottl).
+
+### General Config
+
+> [!NOTE]
+> If you don't know how to write OTTL statements yet, first see [Writing OTTL](#writing-ottl).
+
+```yaml
+transform:
+  error_mode: ignore
+  <trace|metric|log>_statements: []
+```
+
+Only certain OTTL Path prefixes can be used for each signal:
+
+| Signal            | Path Prefix Values                             |
+|-------------------|------------------------------------------------|
+| trace_statements  | `resource`, `scope`, `span`, and `spanevent`   |
+| metric_statements | `resource`, `scope`, `metric`, and `datapoint` |
+| log_statements    | `resource`, `scope`, and `log`                 |
 
-Each statement may have a Where clause that acts as an additional check for whether to execute the statement.
+This means, for example, that you cannot use the Path `span.attributes` within the `log_statements` configuration section.
 
-The transform processor also allows configuring an optional field, `error_mode`, which will determine how the processor reacts to errors that occur while processing a statement.
-The top-level `error_mode` can be overridden at context statements level, offering more granular control over error handling. 
+`error_mode`: determines how the processor reacts to errors that occur while processing a statement.
+If the top-level `error_mode` is not specified, `propagate` will be used.
+The top-level `error_mode` can be overridden at context statements level, offering more granular control over error handling. If context statement `error_mode` is not used, the top-level `error_mode` is used.
 
 | error_mode | description                                                                                                                                 |
 |------------|---------------------------------------------------------------------------------------------------------------------------------------------|
 | ignore     | The processor ignores errors returned by statements, logs the error, and continues on to the next statement.  This is the recommended mode. |
 | silent     | The processor ignores errors returned by statements, does not log the error, and continues on to the next statement.                        |
 | propagate  | The processor returns the error up the pipeline.  This will result in the payload being dropped from the collector.                         |
 
-If not specified, `propagate` will be used.
+### Flat Config
```

Comment:
Would we be able to get away without introducing any new concepts and just say that each item in the `X_statements` list is:

1. "Basic config": A statement
2. "Advanced config": A nested config object within the list that contains a list of statements and options for configuring their execution.

Reply:
Ya those terms work for me, I'll make the updates.

Reply:
updated